### ISSUE TITLE
Invalid model when there is a foreign key that refers to another column in the same table

### ISSUE THREAD

**Issue by IgnatLomkov** [sequelize/sequelize-auto#569]
I have a table of users. 
For an easy understanding of the problem, let's imagine that it has the name **"user"** and the following structure:
**id** - integer - primary key, autoincrement
**inviterId** - integer - bkey with foreign key to **"user"."id"**
**"inviterId"**  - by the referral link of which user this user has registered, accordingly, through SQL, I should be able to get all the user's referrals. But sequelize-auto does not create any field in the typing of the model for this, and as a result, when accessing the user instance using the users key, I get a TypeScript error. I solved this problem by adding users!: user[] to the model class, but it's rather inconvenient to write this every time the models are updated, and therefore I ask you to fix this problem.

**Issue info**
Programming language: TypeScript (Version 4.2.4)
Database: PostgreSQL (Version 12)
Sequelize: 6.6.2 
Sequelize-auto: 0.8.2
pg package: 6.6.2
OS: linux

**Comment by steveschmitt** [comment]
Yes, that makes sense.  One doesn't always want these inverse relationships, but having them in the TypeScript should be harmless, I think.
